ios/RnExecutorch/models/BaseModel.h

```diff
@@ -0,0 +1,14 @@
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+#import "ETModel.h"
+
+@interface BaseModel : NSObject
+{
+@protected
+    ETModel *module;
+}
+
+- (NSArray *)forward:(NSArray *)input shape:(NSArray *)shape inputType:(NSNumber *)inputType;
+- (void)loadModel:(NSURL *)modelURL completion:(void (^)(BOOL success, NSNumber *code))completion;
```

Comment:
Why don't we have the preprocessing and postprocessing functions here?

Reply:
To write an abstract functions in objective-c++ it's required to first write class in pure c++ and then bridge it between c++ and objective-c++. We tried this and it introduced a lot of complexity which in my opinion is unecessary here.

Reply:
If so, then I'm wondering what is even the purpose of this here? How will we use the BaseModel?

Reply:
It's implementing forward and loadModel functions, you don't have to implement them for example in StyleTransferModel. You can just use [self forward: someInput], if we remove BaseModel you will have to implement forward in every model class.